packages/headless/src/features/configuration/configuration-slice.ts

```diff
@@ -72,4 +77,7 @@ export const configurationReducer = createReducer(
       .addCase(setOriginLevel3, (state, action) => {
         state.analytics.originLevel3 = action.payload.originLevel3;
       })
+      .addCase(setAnonymous, (state, action) => {
+        state.analytics.anonymous = action.payload.anonymous;
+      })
```

Comment:
Is there a reason to have two different ways to set this property? If not, I would suggest just using `updateAnalyticsConfiguration`

Reply:
That's similar to all the other analytics property in this file, which follows the same pattern.

You are right that it's not absolutely needed, and all other similar analytics action could in theory be removed ( `disableAnalytics`, `enableAnalytics`, `setOriginLevel2`, `setOriginLevel3`, etc). I'm not sure I remember why they were created in the first place.

We can't do this without a breaking change, however.

Which do you prefer ?

1- I don't add a new function for this, going against what we do for other analytics properties
2- I add one, even if it's repetitive.

Reply:
I prefer to have a single approach, and not propagate patterns unless there is a good reason to do so.
We can also deprecate any redundant actions, log warnings when they are dispatched, and eventually remove them as part of v2.